server/src/main/java/io/crate/execution/engine/aggregation/impl/average/numeric/NumericAverageAggregation.java

```diff
@@ -0,0 +1,339 @@
+                               Version minNodeInCluster,
+                               MemoryManager memoryManager) {
+        ramAccounting.addBytes(INIT_SIZE);
+        return new NumericAverageState(new BigDecimalValueWrapper(BigDecimal.ZERO), 0L);
+    }
+
+    @Override
+    public NumericAverageState iterate(RamAccounting ramAccounting,
+                              MemoryManager memoryManager,
+                              NumericAverageState state,
+                              Input[] args) throws CircuitBreakingException {
+        if (state != null) {
+            BigDecimal value = returnType.implicitCast(args[0].value());
+            if (value != null) {
+                BigDecimal newValue = state.sum.value().add(value);
+                ramAccounting.addBytes(NumericType.sizeDiff(newValue, state.sum.value()));
+                ((BigDecimalValueWrapper) state.sum).setValue(newValue);
+                state.count++;
+            }
+        }
+        return state;
+    }
+
+    @Override
+    public NumericAverageState reduce(RamAccounting ramAccounting,
+                               NumericAverageState state1,
+                               NumericAverageState state2) {
+        if (state1 == null) {
+            return state2;
+        }
+        if (state2 == null) {
+            return state1;
+        }
+        BigDecimal newValue = state1.sum.value().add(state2.sum.value());
+        ramAccounting.addBytes(NumericType.sizeDiff(newValue, state1.sum.value()));
+        ((BigDecimalValueWrapper) state1.sum).setValue(newValue);
+        state1.count += state2.count;
+        return state1;
+    }
+
+    @Override
+    public BigDecimal terminatePartial(RamAccounting ramAccounting, NumericAverageState state) {
+
+        return state.value();
+    }
+
+    @Override
+    public DataType<?> partialType() {
+        return NumericAverageStateType.INSTANCE;
+    }
+
+    @Override
+    public Signature signature() {
+        return signature;
+    }
+
+    @Override
+    public Signature boundSignature() {
+        return boundSignature;
+    }
+
+    @Override
+    public boolean isRemovableCumulative() {
+        return true;
+    }
+
+    @Override
+    public NumericAverageState removeFromAggregatedState(RamAccounting ramAccounting,
+                                                         NumericAverageState previousAggState,
+                                                         Input[] stateToRemove) {
+        if (previousAggState != null) {
+            BigDecimal value = returnType.implicitCast(stateToRemove[0].value());
+            if (value != null) {
+                BigDecimal newValue = previousAggState.sum.value().subtract(value);
+                ramAccounting.addBytes(NumericType.sizeDiff(newValue, previousAggState.sum.value()));
+                previousAggState.count--;
+                ((BigDecimalValueWrapper) previousAggState.sum).setValue(newValue);
```

Comment:
See my previous comment on iterate, implementing `NumnericAverageState.substract()` could be done instead.